lib/rspec/rails/fixture_support.rb

```diff
@@ -13,7 +13,8 @@ module FixtureSupport
         # Monkey patched to avoid collisions with 'let(:name)' in Rails 6.1 and after
         # and let(:method_name) before Rails 6.1.
         def run_in_transaction?
-          use_transactional_tests && !self.class.uses_transaction?(self)
+          current_example_name = (RSpec.current_example && RSpec.current_example.metadata[:description])
```

Comment:
both `@example` and `method_name` works in my Rails project but not here. right now RSpec meta is the most reliable way for me to access the example's name. I know it's ugly so it'd be great if anyone can give me a pointer on this 🙂

Reply:
Wondering what is `self` in this context? Is `self.metadata` or just `metadata` available?

Reply:
calling either one returns 

```
`metadata` is not available from within an example (e.g. an `it` block) or from constructs that 
run in the scope of an example (e.g. `before`, `let`, etc). It is only available on an example group 
(e.g. a `describe` or `context` block).`
```

I do wish methods like `name` or `metadata` is available in an example though.

Reply:
Right.
`@example` seems to be coming from https://github.com/rspec/rspec-core/blob/ea8554afd1a2b63677c6593059fa8f2476181deb/lib/rspec/core/example.rb#L331. It feels that it's an undesirable effect that it is accessible in this scope.

I don't think there's anything wrong with using `current_example` here, even if it's not too elegant.